rero_ils/modules/items/api/api.py

```diff
@@ -247,15 +248,30 @@ def index(self, record):
         document.reindex()
         current_search.flush_and_refresh(DocumentsSearch.Meta.index)
         current_search.flush_and_refresh(HoldingsSearch.Meta.index)
-
-        # check if old holding can be deleted
-        if holding_pid:
-            holding_rec = Holding.get_record_by_pid(holding_pid)
-            try:
-                # TODO: Need to split DB and elasticsearch deletion.
-                holding_rec.delete(force=False, dbcommit=True, delindex=True)
-            except IlsRecordError.NotDeleted:
-                pass
+        # set holding masking for standard holdings
+        new_holdings_pid = extracted_data_from_ref(record['holding']['$ref'])
+        holding = Holding.get_record_by_pid(new_holdings_pid)
+        if holding.get('holdings_type') == 'standard':
+            number_of_unmasked_items = \
+                Item.get_number_masked_items_by_holdings_pid(new_holdings_pid)
+            update_holdings = False
+            if not holding.get('_masked') and not number_of_unmasked_items:
+                holding['_masked'] = True
+                update_holdings = True
+            elif holding.get('_masked') and number_of_unmasked_items:
+                holding['_masked'] = False
+                update_holdings = True
+            if update_holdings:
+                holding = holding.update(
+                    data=holding, dbcommit=True, reindex=True)
```

Comment:
Just like me, you don't have many hairs. So maybe, you can simplify the code fr next read/update :
```python
# masked holding if all items are masked
if number_of_unmasked_items == 0 && not holding.get('_masked'):
    holding['_masked'] = True
    holding.update(data=holding, dbcommit=True, reindex=True)
# unmask holding if some related items are visible
elif number_of_unmasked_items > 0 && holding.get('_masked'):
    holding['_masked'] = False
    holding.update(data=holding, dbcommit=True, reindex=True)
```
1) you skip the third IF statement
2) no need of return value from update call
3) check if reindex is necessary at this step or if another index call is performed after this operation (because we update item or document)

Reply:
`if not number_of_unmasked_items` = `number_of_unmasked_ite<ms == 0`
`if number_of_unmasked_items` = `number_of_unmasked_items == 0`

> check if reindex is necessary at this step or if another index call is performed after this operation (because we update item or document)

checking....